ONSdigital/census-rh-service
# Cr 118 receive case events from rm

# Motivation and Context

Receives CaseEvent type messages and deserialises them into relevant Java objects (CaseEvent objects) and then stores them in a bucket, named case_bucket, within the Google Cloud Platform.

# What has changed

Various Java classes have been added to the package named uk.gov.ons.ctp.integration.rhsvc.message.impl

There is more work to be done but this pull request will enable other developers to make use of what has been done so far.



# How to test?

Use the readme file to firstly set up Google credentials locally. Then run Rabbitmq locally using the command 'docker-compose up -d'. Use the following JSON format to put a CaseEvent type message on the case-outbound-exchange (and use Case.Gateway.binding as the routing key when you publish the message). You can then run the code and should see the message appear in Google Cloud Platform within a bucket named case_bucket NB. To see the bukets you need to select Storage --> Browser (within GCP).

{
  "event" : {
    "type" : "CaseUpdated",
    "source" : "CaseService",
    "channel" : "rm",
    "dateTime" : "2011-08-12T20:17:46.384Z",
    "transactionId" : "c45de4dc-3c3b-11e9-b210-d663bd873d93"
  },
  "payload" : {
    "collectionCase" : {
        "id":"bbd55984-0dbf-4499-bfa7-0aa4228700e9",
        "caseRef":"10000000010",
        "survey":"Census",
        "collectionExerciseId":"n66de4dc-3c3b-11e9-b210-d663bd873d93",
        "sampleUnitRef":"",
        "address": "somewhere-out-there",
        "state":"actionable",
        "actionableFrom":"2011-08-12T20:17:46.384Z"
    }
  }
}
